pygmt/clib/session.py

```diff
@@ -717,7 +717,7 @@ def _check_dtype_and_dim(self, array, ndim):
         """
         # check the array has the given dimension
         if array.ndim != ndim:
-            raise GMTInvalidInput(f"Expected a numpy 1d array, got {array.ndim}d.")
+            raise GMTInvalidInput(f"Expected a numpy {ndim}d array, got {array.ndim}d.")
```

Comment:
For consistency with PR #2252, for the rest please see PR #2307.
```suggestion
            raise GMTInvalidInput(f"Expected a numpy {ndim}-D array, got {array.ndim}-D.")
```